src/pages/detail-page/context/detail-page-provider.tsx

```diff
@@ -27,6 +48,26 @@ export const DetailPageProvider: FC<DetailPageProviderProps> = ({
   >(undefined);
   const [isCollectionNotFound, setIsCollectionNotFound] =
     useState<boolean>(false);
+  const [downloadConditions, _setDownloadConditions] = useState<
```

Comment:
_set instead of setDownloadConditions is because?

Reply:
because i wanted to customize the `setstate` function. So i used `_setxxx` as the `setState` function, and customized it as: `setxxx` which invokes the `_setxxx` to rerender related components.